doc/cprover-manual/modeling-shadow-memory.md

```diff
@@ -0,0 +1,148 @@
+# Shadow Memory
+
+(The original artefact's implementation is outlined in the paper [CBMC-SSM:
+Bounded Model Checking of C Programs with Symbolic Shadow
+Memory](https://dl.acm.org/doi/abs/10.1145/3551349.3559523).)
+
+## Introduction
+
+CBMC contains a component that supports *Symbolic Shadow Memory*. Shadow memory
+allows one to create a parallel memory structure that is maintained by the
+analysis program and that shadows the standard memory of the program. In that
+memory structure, a user can create fields, for which he can later set and
+retrieve values.
+
+By doing so, one can organise his own tracking of metadata related to the code,
+in a way that is then considered by the backend of CBMC during analysis. This
+can be used to implement novel analyses that the CBMC framework itself doesn't
+support - for example, the paper above presents as an example a taint analysis
+implemented with the use of the SSM component described here.
+
+## Usage
+
+A user can interact with the Symbolic Shadow Memory component (from now on
+onwards, SSM) through four CBMC primitives, which allow one to declare shadow
```

Comment:
Suggestion "allow one" -> "support" and splitting into two sentences.
"A user can interact with the Symbolic Shadow Memory component (from now onwards, SSM) through four CBMC primitives. These support the declaration of shadow memory fields, and setting/retrieving their corresponding values:"

Reply:
I changed to "allow" without "one".

Split in 2 sentences.